src/wasm2js.h

```diff
@@ -878,11 +881,6 @@ Ref Wasm2JSBuilder::processFunction(Module* m,
   // sure that everything has a name and it's unique.
   Names::ensureNames(func);
   Ref ret = ValueBuilder::makeFunction(fromName(func->name, NameScope::Top));
-  frees.clear();
-  frees.resize(std::max(Type::i32, std::max(Type::f32, Type::f64)) + 1);
-  temps.clear();
-  temps.resize(std::max(Type::i32, std::max(Type::f32, Type::f64)) + 1);
-  temps[Type::i32] = temps[Type::f32] = temps[Type::f64] = 0;
```

Comment:
Was this resetting only necessary for the asserts below?

Reply:
Good catch, turns out it was not. I did not realize this class reused the instance for all the functions. What made me miss this in the outputs is that any excess temp vars that were added get removed by later optimizations... so it seemed like manually clearing was not needed, but it was. Fixed.